### PR TITLE
Add support for the ESTScan scores matrices (smat) data format

### PR DESCRIPTION
**Pull request by gregvonkuster**

### CODE REVIEW

File: lib/galaxy/datatypes/estscan.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+from galaxy.datatypes.data import get_file_peek, Text
+from galaxy.util import nice_size
+
+log = logging.getLogger(__name__)
+
+MAX_LINE_LEN = 500
+MAX_LINES = 10000
+
+
+class Smat(Text):
```
Comment (bgruening) [inline lib/galaxy/datatypes/estscan.py:13]: Can we have all this in the module where the other general text formats are?

File: lib/galaxy/datatypes/estscan.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                if len(items) != 4:
+                    return rval(False, fh)
+                for item in items:
+                    # Make sure each item is an integer.
+                    if re.match(r"[-+]?\d+$", item) is None:
+                        return rval(False, fh)
+        return rval(True, fh)
+
+
+def rval(rval, fh):
```
Comment (bgruening) [inline lib/galaxy/datatypes/estscan.py:76]: Can we have rval() as subfunction of the sniff function?

Comment (bgruening) [reply lib/galaxy/datatypes/estscan.py:76]: Wait you don't need this function at alle. `with` is taking care of closing the handle for you. This is a python feature.

File: lib/galaxy/datatypes/estscan.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            if not line.startswith('FORMAT'):
+                if line.find('\t') >= 0:
+                    # Smat files are not tabular.
+                    return rval(False, fh)
+                items = line.split()
+                if len(items) != 4:
+                    return rval(False, fh)
+                for item in items:
+                    # Make sure each item is an integer.
+                    if re.match(r"[-+]?\d+$", item) is None:
```
Comment (bgruening) [inline lib/galaxy/datatypes/estscan.py:71]: Can you use isdiget() here?

Comment (bgruening) [reply lib/galaxy/datatypes/estscan.py:71]: [item for itme in item if item.isdigit()]

Comment (gregvonkuster) [reply lib/galaxy/datatypes/estscan.py:71]: Won't work for negative integers.
```
>>> i="-1"
>>> i.isdigit()
False
```

File: lib/galaxy/datatypes/estscan.py (hunk 2 above)
Comment (gregvonkuster) [reply lib/galaxy/datatypes/estscan.py:76]: Ok, I'll fix.  Thanks!

File: lib/galaxy/datatypes/estscan.py (hunk 1 above)
Comment (gregvonkuster) [reply lib/galaxy/datatypes/estscan.py:13]: Do you mean data.py?  There are a lot of code files whose classes subclass from data.Text: interval.py, molecules.py, mothur.py, msa.py, proteomics.py, etc.  I will move it to data.py if you feel that is where it should go.

Comment (gregvonkuster) [reply lib/galaxy/datatypes/estscan.py:13]: Ah, text.py obviously.  I'll move it there.

**Comment by gregvonkuster** [comment]
moving to text.py in another PR.

**Pull request closed without merging**